Jointify/Model/PoseMachineLearning/PoseNet.swift

```diff
@@ -0,0 +1,332 @@
+        } else {
+            print("Error. At least one vector is of size 0")
+        }
+
+        return innerAngle
+    }
+
+    /// Vertically flips and draws the given image.
+    ///
+    /// - parameters:
+    ///     - image: The image to draw onto the context (vertically flipped).
+    ///     - cgContext: The rendering context.
+    private func drawImage(image: CGImage, in cgContext: CGContext) {
+        cgContext.saveGState()
+        // The given image is assumed to be upside down; therefore, the context
+        // is flipped before rendering the image.
+        cgContext.scaleBy(x: 1.0, y: -1.0)
+        // Render the image, adjusting for the scale transformation performed above.
+        let drawingRect = CGRect(x: 0, y: -image.height, width: image.width, height: image.height)
+        cgContext.draw(image, in: drawingRect)
+        cgContext.restoreGState()
+    }
+ 
+    /// Draws a line between two joints.
+    ///
+    /// - parameters:
+    ///     - parentJoint: A valid joint whose position is used as the start position of the line.
+    ///     - childJoint: A valid joint whose position is used as the end of the line.
+    ///     - cgContext: The rendering context.
+    private func drawLine(from parentJoint: Joint,
+                          to childJoint: Joint,
+                          in cgContext: CGContext) {
+        cgContext.setStrokeColor(Constants.segmentColor.cgColor)
+        cgContext.setLineWidth(Constants.segmentLineWidth)
+
+        cgContext.move(to: parentJoint.position)
+        cgContext.addLine(to: childJoint.position)
+        cgContext.strokePath()
+    }
+
+    /// Draw a circle in the location of the given joint.
+    ///
+    /// - parameters:
+    ///     - circle: A valid joint whose position is used as the circle's center.
+    ///     - cgContext: The rendering context.
+    private func drawJoints(circle joint: Joint, in cgContext: CGContext) {
+        cgContext.setFillColor(Constants.jointColor.cgColor)
+
+        let rectangle = CGRect(x: joint.position.x - Constants.jointRadius, y: joint.position.y - Constants.jointRadius,
+                               width: Constants.jointRadius * 2, height: Constants.jointRadius * 2)
+        cgContext.addEllipse(in: rectangle)
+        cgContext.drawPath(using: .fill)
+    }
+    
+    // Helper function to convert CIImage to CGImage
+    private func convertCIImageToCGImage(inputImage: CIImage) -> CGImage? {
+        let context = CIContext(options: nil)
+        return context.createCGImage(inputImage, from: inputImage.extent)
+    }
+    
+    /// Draw a circle in the location of the given joint.
+    /// Returns an UIImage with the specified edges and joints drawn on it.
+    ///
+    /// - parameters:
+    ///     - image: The image to be analysed.
+    func predict (_ image: UIImage) -> UIImage {
+        
+        // Add alternative image which is definitely shipped once
+        // swiftlint:disable force_unwrapping
```

Comment:
@BergmuellerNiklas this is how I fixed this annoying UIImage that has to be returned in the guard statements. Nevertheless, this one force unwrap is as bad as hundreds of them as there just needs to be a problem with transferring this file into the app and it will crash without a warning.

**The proper way would actually be to make this an error throwing function, sending the problems that can arise in the analysis up to the parent view so it can take care of displaying an alert to the user 😏**